portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/SearchAPI.testcase

```diff
@@ -81,19 +76,134 @@ definition {
+	@priority = "3"
+	test RequestKeywordsUsingNoParameters {
+		property test.name.skip.portal.instance = "SearchAPI#RequestKeywordsUsingNoParameters";
+
+		var keyword1 = "without optional parameters 1";
+		var keyword2 = "without optional parameters 2";
+
+		var keywordList = "${keyword1},${keyword2}";
+
+		task ("Go to Server Administration") {
+			ServerAdministration.openServerAdmin();
+		}
+
+		task ("Get Faro Backend URL") {
+			var faroBackendURL = ACApi.getSearchHeaderValues(header = "liferayAnalyticsFaroBackendURL");
+		}
+
+		task ("Get OSB-Asah-Faro-Backend-Security-Signature") {
+			var osbAsahFaroBackendSS = ACApi.getSearchHeaderValues(header = "liferayAnalyticsFaroBackendSecuritySignature");
+		}
+
+		task ("Get OSB-Asah-Project-Id") {
+			var osbAsahProjectId = ACApi.getSearchHeaderValues(header = "liferayAnalyticsProjectId");
+		}
+
+		task ("Go to Site Page") {
+			ACUtils.navigateToSitePage(
+				pageName = "Home",
+				siteName = "guest");
+		}
+
+		task ("Type keyword in search bar") {
+			for (var keyword : list "${keywordList}") {
+				SearchPortlets.searchEmbedded(searchTerm = "${keyword}");
+			}
+		}
+
+		task ("Run curl command without optional parameters to get keyword results") {
+			ACApi.getSearchKeywords(
+				faroBackendURL = "${faroBackendURL}",
+				osbAsahFaroBackendSS = "${osbAsahFaroBackendSS}",
+				osbAsahProjectId = "${osbAsahProjectId}");
+		}
+
+		task ("Assert response returns keyword") {
+			ACApi.assertSearchResponse(expected = "${keyword1}");
+		}
+
+		task ("Assert response returns keyword") {
+			ACApi.assertSearchResponse(
+				expected = "${keyword2}",
+				keywordPosition = "1");
+		}
+	}
+
+	@description = "Feature ID: LRAC-12058 | Automation ID: LRAC-12146 | Test Summary: Fetch keywords with size parameter "
+	@priority = "3"
+	test RequestKeywordsUsingSize {
+		var keyword1 = "with size parameters 1";
+		var keyword2 = "with size parameters 2";
+		var keyword3 = "with size parameters 3";
+		var parameters = "size=2&sort=lastModifiedDate,desc";
+		var keywordList = "${keyword1},${keyword2},${keyword3}";
+		var size = "2";
+
+		task ("Go to Server Administration") {
+			ServerAdministration.openServerAdmin();
+		}
+
+		task ("Get Faro Backend URL") {
+			var faroBackendURL = ACApi.getSearchHeaderValues(header = "liferayAnalyticsFaroBackendURL");
+		}
+
+		task ("Get OSB-Asah-Faro-Backend-Security-Signature") {
+			var osbAsahFaroBackendSS = ACApi.getSearchHeaderValues(header = "liferayAnalyticsFaroBackendSecuritySignature");
+		}
+
+		task ("Get OSB-Asah-Project-Id") {
+			var osbAsahProjectId = ACApi.getSearchHeaderValues(header = "liferayAnalyticsProjectId");
+		}
+
+		task ("Go to Site Page") {
+			ACUtils.navigateToSitePage(
+				pageName = "Home",
+				siteName = "guest");
+		}
+
+		task ("Type keyword in search bar") {
+			for (var keyword : list "${keywordList}") {
+				SearchPortlets.searchEmbedded(searchTerm = "${keyword}");
```

Comment:
Hey Vini, what do you think instead of creating a list with your keywords (line 98) you can just pass the keywords in the for loop and delete the list:
`for (var keyword : list "${keyword1},${keyword2},${keyword3}")`
what do you think?